pallets/oracle-data-collection/src/lib.rs

```diff
@@ -0,0 +1,496 @@
+		KeyNotRegistered,
+
+		/// Collection size reached
+		MaxCollectionSize,
+
+		/// Collection size reached
+		NoOracleCollectionChangeId,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Propose an update in the feeders associated to an specific key.
+		/// The collection will only be modified once
+		/// [`Pallet::apply_update_feeders`] be called.
+		#[pallet::weight(T::WeightInfo::propose_update_feeders(T::MaxFeedersPerKey::get()))]
+		#[pallet::call_index(0)]
+		pub fn propose_update_feeders(
+			origin: OriginFor<T>,
+			collection_id: T::CollectionId,
+			key: T::OracleKey,
+			feeders: BoundedVec<T::FeederId, T::MaxFeedersPerKey>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				T::IsAdmin::check((who, collection_id)),
+				Error::<T>::IsNotAdmin
+			);
+
+			transactional::with_transaction(|| {
+				let result = Self::update_feeders(collection_id, key, feeders.clone());
+
+				// We do not want to apply the mutation,
+				// only check if there is no error in applying it
+				TransactionOutcome::Rollback(result)
+			})?;
+
+			T::ChangeGuard::note(collection_id, Change::Feeders(key, feeders).into())?;
+
+			Ok(())
+		}
+
+		/// Apply an change previously proposed by
+		/// [`Pallet::propose_update_feeders`] if the conditions to get it ready
+		/// are fullfilled
+		///
+		/// This call is permissionless
+		#[pallet::weight(T::WeightInfo::apply_update_feeders(T::MaxFeedersPerKey::get()))]
+		#[pallet::call_index(1)]
+		pub fn apply_update_feeders(
+			origin: OriginFor<T>,
+			collection_id: T::CollectionId,
+			change_id: T::Hash,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let Change::Feeders(key, feeders) = T::ChangeGuard::released(collection_id, change_id)?
+				.try_into()
+				.map_err(|_| Error::<T>::NoOracleCollectionChangeId)?;
+
+			Self::update_feeders(collection_id, key, feeders.clone())?;
+
+			Self::deposit_event(Event::<T>::UpdatedFeeders {
+				collection_id,
+				key,
+				feeders,
+			});
+
+			Ok(())
+		}
+
+		/// Update the collection, doing the aggregation for each key in the
+		/// process.
+		///
+		/// This call is permissionless
+		#[pallet::weight(T::WeightInfo::update_collection(
+			T::MaxFeedersPerKey::get(),
+			T::MaxCollectionSize::get(),
+		))]
+		#[pallet::call_index(2)]
+		pub fn update_collection(
+			origin: OriginFor<T>,
+			collection_id: T::CollectionId,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let values = Keys::<T>::iter_key_prefix(collection_id)
+				.filter_map(|key| match Self::get(&key, &collection_id) {
+					Ok(value) => Some(Ok((key, value))),
+					Err(err) if err == Error::<T>::KeyNotInCollection.into() => None,
+					Err(err) => Some(Err(err)),
+				})
+				.collect::<Result<BTreeMap<_, _>, _>>()?;
+
+			let collection =
+				BoundedBTreeMap::try_from(values).map_err(|()| Error::<T>::MaxCollectionSize)?;
+
+			let len = collection.len();
+
+			Collection::<T>::insert(collection_id, collection);
```

Comment:
Q: It is intended to overwrite any existing storage entry, correct?

Reply:
Yes, that is the intention. Thanks for checking it